core/src/main/scala/internal/NettySupport.scala

```diff
@@ -444,72 +436,65 @@ private[http] object NettySupport {
     // When the channel is closed we push None to the message
     // queue, to indicate the End Of Stream. We also push None
     // to the content queue to force incomplete stream to finish.
-    lazy val closed: Task[Unit] = channel.closeFuture.toTask.flatMap { _ =>
-      (for {
+    lazy val closed: IO[Unit] = channel.closeFuture.toIO.flatMap { _ =>
+      for {
         _ <- messages.enqueue1(None)
         _ <- content.enqueue1(None)
-      } yield ())
+      } yield ()
     }
 
     def isOpen: Boolean = channel.isOpen
-    def close: Task[Unit] = Task.delay(if(channel.isOpen) channel.close())
+    def close: IO[Unit] = IO(if(channel.isOpen) channel.close())
 
     // Read one HTTP message along with its content stream. The
     // content stream must be read before the next message to be
     // available.
-    def read(): Task[(HttpMessage,Stream[Task,Byte])] =
+    def read: IO[(HttpMessage, Stream[IO, Byte])] =
       messages.dequeue1.flatMap {
         case Some((message, true)) =>
           for {
-            readers <- async.semaphore[Task](1) // Keep track of the number of readers
-            eosReached <- async.signalOf[Task,Boolean](false) // Keep track of the End Of Stream
+            readers <- async.semaphore[IO](1)
+            eosReached <- async.signalOf[IO, Boolean](false)
             messageStream =
-              Stream.
-                // The content stream can be read only once
-                eval(readers.tryDecrement).flatMap {
-                  case false =>
-                    Stream.fail(Error.StreamAlreadyConsumed)
-                  case true =>
-                    contentStream.
-                      // We read the queue until a None, that mark
-                      // the content stream end.
-                      evalMap { chunk => eosReached.set(chunk.isEmpty).map(_ => chunk) }.
-                      takeWhile(_.isDefined).
-                      // The stream of bytes
-                      flatMap(chunk => Stream.chunk(chunk.get)).
-                      // When user code finished to consume this stream, we need
-                      // to drain the remaining content if the eos has not been
-                      // reached yet.
-                      onFinalize(for {
-                        fullyRead <- eosReached.get
-                        _ <- if(fullyRead) Task.now(()) else contentStream.takeWhile(_.isDefined).drain.run
-                        _ <- if(client) permits.increment else permits.decrement
-                      } yield ())
-                }
+              Stream.eval(readers.tryDecrement).flatMap {
```

Comment:
Sorry again, when I refactored that, I made a mental note to put those comments back, but it slipped my mind.